django-stubs/db/models/fields/__init__.pyi

```diff
@@ -205,7 +205,7 @@ class Field(RegisterLookupMixin, Generic[_ST, _GT]):
         form_class: type[forms.Field] | None = None,
         choices_form_class: type[forms.ChoiceField] | None = None,
         **kwargs: Any,
-    ) -> forms.Field: ...
+    ) -> forms.Field | None: ...
```

Comment:
hmmm so this is triggering a bunch of false positives after upgrading django-stubs

it seems to me that `django.db.models.fields.Field.formfield` *always* returns a `Field` so this annotation is wrong (?) and subclasses should override it to be `| None` if applicable (?) (even though it would violate substitutability -- but such is django)

Reply:
Did you read the associated ticket and docs change? Any `Field` subclass may return `None` from `formfield()` to remove it from forms.

In Django itself, `AutoFieldMixin` and `OneToOneField` use this feature. I have used this feature at least once in my MySQL package: https://github.com/adamchainz/django-mysql/blob/1ba0471c06bf7e3b47a1ee2c544fc3738020611e/src/django_mysql/models/fields/dynamic.py#L295-L299

Adding the `None` option the base class fixes substitutability, no?

Reply:
yes but `fields.Field` does not return None, and my subclass of `fields.Field` inherits the same implementation which does not return `None` -- and now the type checker is mad about that

Reply:
for example this: https://github.com/adamchainz/django-jsonfield/blob/939c5a7b8c5414d2b63aa141899c9059b8ce79ec/jsonfield/tests/test_fields.py#L55-L57

Reply:
Sure.. I'm happy adding `assert formfield is not None` to such tests. That's just type-hinted life, in my experience.